errors.go

```diff
@@ -118,92 +119,112 @@ func (p *Error) LogMetadata() map[string]string {
 // New creates a new error for you. Use this if you want to pass along a custom error code.
 // Otherwise use the handy shorthand factories below
 func New(code string, message string, params map[string]string) *Error {
-	return errorFactory(code, message, params)
+	return errorFactory(code, message, params, nil)
 }
 
 // Wrap takes any error interface and wraps it into an Error.
 // This is useful because an Error contains lots of useful goodies, like the stacktrace of the error.
 // NOTE: If `err` is already an `Error`, it will add the params passed in to the params of the Error
-func Wrap(err error, params map[string]string) error {
-	return WrapWithCode(err, params, ErrInternalService)
+func Wrap(err error, params map[string]string, messages ...string) error {
+	return WrapWithCode(err, params, ErrInternalService, messages...)
 }
 
 // WrapWithCode wraps an error with a custom error code. If `err` is already
 // an `Error`, it will add the params passed in to the params of the error
-func WrapWithCode(err error, params map[string]string, code string) error {
+func WrapWithCode(err error, params map[string]string, code string, messages ...string) error {
 	if err == nil {
 		return nil
 	}
-	switch err := err.(type) {
+
+	// We're using a variadic function here but only a single message is actually used.
+	// This is designed to ensure backwards compatability for usages without a message.
```

Comment:
I'm unsure about whether this is a worthwhile thing to do... It allows for an easy migration but it does make the interface a little strange. We could get around this by a) making a breaking change or b) introducing a new method that supports a `message` argument.

Reply:
I quite like the idea of a new method if it feels more natural. We could also accept `map[string]interface{}` to align with slog changes too. Maybe `terrors.WithContext(err error, message string, data map[string]interface{})` or similar?